### PR TITLE
.ron > .ronn

### PR DESCRIPTION
**Pull request by CAD97**
The man page authoring tool [ronn] [hasn't been called ron since 2010](https://rubygems.org/gems/ron/versions/0.4).

Today on GitHub, there are _far_ more [.ronn] files than [.ron] files, and most of the (unique) .ron files appear to be the [RON] data format rather than Ronn markup.

This change seems like it would benefit both languages: the many .ronn files will now get picked up as markup and rendered in the GitHub preview, and the RON data files will show up in a monospace whitespace-preserving presentation, instead of being mangled as GitHub tries to reinterpret the unrelated data as markdown. [In fact, there's an open issue on RON about working around this GitHub behavior.](https://github.com/ron-rs/ron/issues/85)

I believe the correct move is to change this extension: the line in question has been touched twice: once [in 2011] back at the beginning of history (of this repo), and once [in 2016] when the YAML was changed from using unquoted strings to quoted strings.

  [Ronn]: <http://rtomayko.github.io/ronn/>
  [.ronn]: <https://github.com/search?utf8=%E2%9C%93&q=extension%3Aronn+NOT+%3Cdata%3E&type=>
  [.ron]: <https://github.com/search?utf8=%E2%9C%93&q=extension%3Aron+NOT+%3Cdata%3E&type=>
  [RON]: <https://github.com/ron-rs/ron>

  [in 2011]: <https://github.com/github/linguist/commit/37243ec6bed3e697beeb874fc94efb9e42ca7f8c>
  [in 2016]: <https://github.com/github/linguist/commit/e0d890240bb5bc6ded6280cee5c506791145926a>

## Checklist:

- [X] **I am associating a language with a new file extension.**
  - [X] The new extension is used in hundreds of repositories on GitHub.com
    - Search results for each extension:
      - https://github.com/search?utf8=%E2%9C%93&type=Code&ref=searchresults&q=extension%3Aronn+NOT+nothack
  - [ ] I have included a real-world usage sample for all extensions added in this PR:
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

- [X] **I am fixing a misclassified language**
  - [ ] I have included a new sample for the misclassified language:
    - Sample source(s):
      - [URL to each sample source, if applicable]
    - Sample license(s):
  - [ ] I have included a change to the heuristics to distinguish my language from others using the same extension.

### CODE REVIEW

File: lib/linguist/languages.yml [hunk 1]
```diff
@@ -2659,7 +2659,7 @@ Markdown:
   - ".mkd"
   - ".mkdn"
   - ".mkdown"
-  - ".ron"
+  - ".ronn"
```
[the remark this answers is not in the document]
Comment (CAD97) [reply lib/linguist/languages.yml]: I just added ronn's own ronn man pages, along with those of bunyan.

**Comment by CAD97** [comment]
On classifying `.ron`: I apparently have too much free time; here's a classification of all [339 code results](https://github.com/search?utf8=%E2%9C%93&q=extension%3Aron+NOT+nothack&type=) for `extension:ron NOT nothack`:

<details><summary>ronn - true positive</summary>

- mustache/mustache - mustache.1.ron
  - plus2/common_mob
  - guinascimento/sac
  - zakk4223/shmups-hs
  - hoverbird/Storying
  - ajacksified/restful-clients-in-rails-demo
  - mattgraham/play-graham
  - supersym/workflow.templates
  - kmcminn/rails_survey
  - mike-eddie/WessexCCP
  - lorint/MongoApartment
  - tklee/jrubystack-1.7.5-0
  - leticiamatos/pcj_old
  - usmschuck/canvas
  - soilforlifeforms/forms
  - soilforlifeforms/forms [yes, twice]
  - objectkuan/osv-dde
  - khrtz/rails_vote_ranking
  - pauldetarse/intestellarserverone
  - khrtz/MonsterHunt
  - roshanatl/logstash-5.0.1
  - rosberglinhares/IISLogMonitor
  - diegojesus/logstash
  - nshetgao/logstash
  - ayushinigam/ELK
  - patuchov/my-jelk-stack
  - iremgokceyildirim/discourse_narrative_matching
  - iremgokceyildirim/discourse_narrative_matching [yes, twice]
  - abhayshukla04/elastic
  - casperwitting/Bigdash
  - year112/Logstash
  - Ruoyu111/DataVisualizationProject
  - miguel-rodriguez/Docker-ELK-Alfresco-Monitoring
  - taliesins/LogStashService
  - wgltony/ReactRealTimeNewsScrapingandRecommendationSystem-GraphiteSystemMonitor
  - RS_ACCELERATE/tools
  - abisheksurya/Community_Dashboard
  - dongbo0737/logstash
  - tiwarisahil91/Twitter-Sentiment-Analysis-of-Soccer-Teams
  - tallestross/devops
  - kyledevine/bet
  - BGGANF/logstash
  - samuel100u/logstash
  - EduJGURJC/etm-logstash
  - Wshid/molit_HAB
  - giorgio94c/elk-quickstart
  - Wshid/metro_analysis
  - SilviaTian/code_jiukang
  - SilviaTian/code_jiukang [yes, twice]
  - Tweeteeeee/capstone
  - kl4314/bigdata-analysis
  - fralvarezcalvo/elk-config
  - sowa-io/ELK-sowa.io_linux_amd64
  - yuexiangqianlong/daka
  - DILIPCHIRU/Logstash
  - ParserKurt/jdbc-importer
  - disigma-dev/app
  - clbustos/buhos-windows-tk
  - MaherBTA/logstash
  - JonCon732/Tweet-Farming
  - EGC1995/Logstash-Kibana-Elastic
  - DeclanBU/OOP2-Project-2014
  - rabmarjan/LogAnalyzer
  - shruthipinnamwar/SleepAnalysis_Fixers
  - dashjuvi/logstash
  - whhub/ELK - mustache.1.ron
- mustache/mustache - mustache.5.ron
  - plus2/common_mob
  - guinascimento/sac
  - zakk4223/shmups-hs
  - r14c/wildebai
  - hoverbird/Storying
  - ajacksified/restful-clients-in-rails-demo
  - mattgraham/play-graham
  - supersym/workflow.templates
  - kmcminn/rails_survey
  - mike-eddie/WessexCCP
  - lorint/MongoApartment
  - tklee/jrubystack-1.7.5-0
  - leticiamatos/pcj_old
  - usmschuck/canvas
  - soilforlifeforms/forms
  - soilforlifeforms/forms [yes, twice]
  - objectkuan/osv-dde
  - khrtz/rails_vote_ranking
  - pauldetarse/intestellarserverone
  - khrtz/MonsterHunt
  - roshanatl/logstash-5.0.1
  - rosberglinhares/IISLogMonitor
  - diegojesus/logstash
  - nshetgao/logstash
  - ayushinigam/ELK
  - patuchov/my-jelk-stack
  - iremgokceyildirim/discourse_narrative_matching
  - iremgokceyildirim/discourse_narrative_matching [yes, twice]
  - abhayshukla04/elastic
  - casperwitting/Bigdash
  - year112/Logstash
  - Ruoyu111/DataVisualizationProject
  - miguel-rodriguez/Docker-ELK-Alfresco-Monitoring
  - taliesins/LogStashService
  - wgltony/ReactRealTimeNewsScrapingandRecommendationSystem-GraphiteSystemMonitor
  - RS_ACCELERATE/tools
  - abisheksurya/Community_Dashboard
  - dongbo0737/logstash
  - tiwarisahil91/Twitter-Sentiment-Analysis-of-Soccer-Teams
  - tallestross/devops
  - kyledevine/bet
  - BGGANF/logstash
  - samuel100u/logstash
  - EduJGURJC/etm-logstash
  - Wshid/molit_HAB
  - giorgio94c/elk-quickstart
  - Wshid/metro_analysis
  - SilviaTian/code_jiukang
  - SilviaTian/code_jiukang [yes, twice]
  - Tweeteeeee/capstone
  - kl4314/bigdata-analysis
  - fralvarezcalvo/elk-config
  - sowa-io/ELK-sowa.io_linux_amd64
  - yuexiangqianlong/daka
  - DILIPCHIRU/Logstash
  - ParserKurt/jdbc-importer
  - disigma-dev/app
  - clbustos/buhos-windows-tk
  - MaherBTA/logstash
  - JonCon732/Tweet-Farming
  - EGC1995/Logstash-Kibana-Elastic
  - DeclanBU/OOP2-Project-2014
  - rabmarjan/LogAnalyzer
  - shruthipinnamwar/SleepAnalysis_Fixers
  - dashjuvi/logstash
  - whhub/ELK - mustache.1.ron
- kennethreitz-archive/kennethreitz.github.com - kennethreitz.1.ron
- [I can't find the origin] - functools.1.ron
  - abesto/YASIOC
  - christkv/repl
- arver/arver - arver.5.ron
- trivera/burrito - burrito.1.ron
- Jwata/gist-old - gist.1.ron
  - higanworks/chef-with-ruby_precise-x86_64
- bufferings/bufferings.github.io - mustache_ja.5.ron
- AtelierConvivialite/webtranslateit - wti.1.ron
  - nyaawira/nt_aws
  - nyaawira/nt_aws [yes, twice]
  - tjquinn1/gidget_2
  - fcmichigan/Marketplace
  - dennisFCWM/Marketplace
- gaenserich/hostsblock - hostsblock-urlcheck.8.ron
- gaenserich/hostsblock - hostsblock.8.ron
- efornara/jbit - jbit.ron
- dawi2332/realpath - realpath.1.ron
- krommydakis/schleuder-master - schleuder-api-daemon.8.ron
- krommydakis/schleuder-master - schleuder.8.ron
- krommydakis/schleuder-cli-master - schleuder-cli.8.ron
- krommydakis/schleuder-master-321 - schleuder-api-daemon.8.ron
- krommydakis/schleuder-master-321 - schleuder.8.ron
- schleuder/schleuder3 - schleuder-api-daemon.8.ron
- schleuder/schleuder3 - schleuder.8.ron
</details>

<details><summary>Rusty Object Notation - false positive</summary>

- kvark/plane-splitter - poly.ron
- Rhuagh/amethyst-experiments - bindings.ron
- Rhuagh/amethyst-experiments - bindings.ron [distinct]
- ron-rs/ron - example.ron
- rsaarelm/magog - sprint.ron
- lukebitts/pipe-away - Config.ron
- minecrawler/ron-debug - config.ron
- ozkriff/zemeroth_assets - settings.ron
- alerdenisov/amethyst-experiment - display.ron
- 171230839/myEmethyst - config.ron
- 171230839/myEmethyst - config.ron [distinct]
- 171230839/myEmethyst - config.ron [distinct]
- 171230839/myEmethyst - config.ron [distinct]
- 171230839/myEmethyst - config.ron [distinct]
- 171230839/myEmethyst - display.ron
- 171230839/myEmethyst - display.ron [distinct]
- 171230839/myEmethyst - display.ron [distinct]
- 171230839/myEmethyst - input.ron
- Rhaugh/rebindit-rs - complex.ron
- Rhaugh/rebindit-rs - simple.ron
- Reificator/amethyst-hello-world - display_config.ron
- Reificator/amethyst-hello-world - input.ron
- Reificator/gem-assault - display_config.ron
- jonathanrlouie/amethyst-tilemap-demo - display.ron
- minecrawler/botto-chan - config.ron
- jswrenn/laserstorm - display_config.ron
- jswrenn/laserstorm - input.ron
- Systemcluster/Dust - person.ron
- Mistodon/config_struct - config.ron
- Daepso/rusty-snake - input.ron
- Daepso/rusty-snake - display.ron
- amethyst/tools - index.ron
- amethyst/tools - display_config.ron
- amethyst/amethyst - mesh.ron
- amethyst/amethyst - config.ron
- amethyst/amethyst - display_config.ron
- amethyst/amethyst - display_config.ron [distinct]
- amethyst/amethyst - display_config.ron [distinct]
- amethyst/amethyst - display_config.ron [distinct]
- amethyst/amethyst - display.ron
- amethyst/amethyst - display_config.ron [distinct]
- amethyst/amethyst - display_config.ron [distinct]
- amethyst/amethyst - display.ron [distinct]
- amethyst/amethyst - display.ron [distinct]
- amethyst/amethyst - display_config.ron [distinct]
- jonathanrlouie/amethyst-tetris - display.ron
- trsoluti/space_shooter - display_config.ron
- trsoluti/space_shooter - game_config.ron
- trsoluti/space_shooter - input.ron
- amethyst/amethyst - display_config.ron
- mozilla/gecko-dev - example.ron
- mozilla/gecko - example.ron
- gfx-rs/gfx - ci.ron
- gfx-rs/gfx - local.ron
- gfx-rs/gfx - transfer.ron
- amethyst/amethyst - config.ron
- amethyst/amethyst - display.ron
- amethyst/amethyst - input.ron
- cmsd2/gamediary - display_config.ron
- xeonchen/gecko - example.ron
- huhlig/shadowstar - display.ron
- Moggers/openrlfb - display.ron
- matklad/libsyntax2 - grammar.ron
- redox-os/orbgame - game.ron
- jojolepro/world-digger - input.ron
- amethyst/amethyst - display_config.ron
- amethyst/amethyst - input.ron
- amethyst/amethyst - input.ron [distinct]
- azriel91/amethyst_menu_experiment - display_config.ron
- jojolepro/TaikoCopy - codes.ron
  - jojolepro/TaikoCopy [yes, twice]
- jojolepro/TaikoCopy - paths.ron
- jojolepro/TaikoCopy - cfg.ron
- jojolepro/TaikoCopy - player_settings.ron
- jojolepro/TaikoCopy - input.ron
- jojolepro/TaikoCopy - lang.ron
- Demivan/game - display_config.ron
- ozkriff/zemeroth_assets - objects.ron
- jojolepro/world-digger - display_config.ron
- jaemk/transfer - config.ron
- vstepchik/mold - bindings.ron
- vstepchik/mold - display_config.ron
- Moggers/amethyst_servo - display.ron
- bnyu/game - display_config.ron
- e-matteson/pipit-keyboard - 0_quick_tour.ron
- e-matteson/pipit-keyboard - 1_bottom_row.ron
- e-matteson/pipit-keyboard - 2_top_row.ron
- e-matteson/pipit-keyboard - 3_double_letters.ron
- e-matteson/pipit-keyboard - 4_right_double_letters.ron
- e-matteson/pipit-keyboard - 5_review_letters.ron
- e-matteson/pipit-keyboard - 6_punctuation_and_capitals.ron
- e-matteson/pipit-keyboard - 7_numbers.ron
- e-matteson/pipit-keyboard - 8_all_symbols.ron
- e-matteson/pipit-keyboard - 9_Alice_in_Wonderland.ron
- e-matteson/pipit-keyboard - 10_code.ron
- rust-dsp/cli-vst-host - audio_config.ron
- JadeSnail/game - config.ron
- JadeSnail/game - display.ron
- thepowersgang/fury3clone - fgl.ron
- jupart/skirm - skirmers.ron
- jupart/skirm - weapons.ron
- derekdreery/shootem - display_config.ron
- derekdreery/shootem - input.ron
- callym/opalite - Opalite.ron
- Moggers/amethyst_servo - display_config.ron
  - Moggers/amethyst_servo [yes, twice]
- amethyst/amethyst - bindings_config.ron
- amethyst/amethyst - display_config.ron
- gfx-rs/gfx - basic.ron
- gfx-rs/gfx - compute.ron
- callym/opalite - Oalite.ron [distinct]
- amethyst/amethyst - display_config.ron
- amethyst/amethyst - srites.ron
- nikomatsakis/ndm-cards - .mathema-v1.ron
- CAD97/ron-display - demo.ron
- JadeSnail/game - input.ron
- jojolepro/TaikoCopy - config.ron

[The reason for so many like-named files is that the Amethyst game engine
uses programmer-facing RON configuration for your project]
</details>

<details>Some XML-like - false positive</summary>
- roland-wilhelm/iot - RTOSDemo.uvgui.Ron
- kuldipsciter/S310_SDK_5_2_0_51422_DEVKIT - DEVKIT_S310_nRF51422.uvgui.RON
- x4121/Ereignisgesteuerte - blinky.uvgui.Ron
  - x4121/Ereignisgesteuerte [yes, twice]
- x4121/Ereignisgesteuerte - bomb4.uvgui.Ron
  - x4121/Ereignisgesteuerte [yes, twice]
- x4121/Ereignisgesteuerte - bomb2.ubgui.Ron
- trampi/eds-ss15 - comp.uvgui.Ron
- bubnikv/omniasdrbasicfirmware - bootloader.cyprj.RON

</details>

<details>Unknown - false positive</summary>

- [https://gitlab.flux.utah.edu/emulab/emulab-stable/] - supfile.ron
  - senseisimple/emulab-stable
  - mahyuddin/emulab-stable
  - vnaybhat/emulab
  - nmc-probe/emulab-nome
  - nmc-probe/emulab-portal
  - nmc-probe/emulab-diskless
- roger-s26/HelloMydroid - tmux.cong.ron
- BenedictC/RON - readme.ron
- Empty007/morfologia - prof.c.ron
- plotozhu/jxdemo - index.ron
- sscotth/file-icon-supplement - markdown.ron [EMPTY; unclassified]
- BackupTheBerlios/sorcerer-svn - tesseract-ocr.ron
- rylectro/atom - markdown.ron [EMPTY; unclassified]
- Tiim/.atom - markdown.ron [EMPTY; unclassified]
- zhangxiaoyu11/mAMBER - Makefile.ron
  - tianhe2/mAMBER - Makefile.ron
- G4te-Keep3r/HowdyHackers - file.ron
- GregTrevellick/FilesForEveryExtension - GitIgnore.RON
- abisheksurya/Community_Dashboard - ._mustache.1.ron [BINARY]
- abisheksurya/Community_Dashboard - ._mustache.1.ron [BINARY]
- ibawolf/pevagrant - Vagrantfile.ron [RUBY set by emacs line]
- NPoe/chunker - PARAM.ron
- gritzko/monorepko - 00-rga-basic.ron
- gritzko/monorepko - 000-go.ron
- gritzko/monorepko - 001-conn.ron
- gritzko/monorepko - 002-hs.ron
- gritzko/monorepko - 003-calendar-clock.ron
- gritzko/monorepko - 004-query.ron
- gritzko/monorepko - 005-push.ron
- gritzko/swarm - 000-go.ron
- gritzko/swarm - 001-conn.ron
- gritzko/swarm - 002-hs.ron
- gritzko/swarm - 003-calendar-clock.ron
- gritzko/swarm - 004-query.ron
- gritzko/swarm - 005-push.ron
- gritzko/swarm - 006-lwwset.ron
- gritzko/swarm - 007-pending.ron
- gritzko/swarm - 008-setadd.ron
- gritzko/swarm - 009-pending.ron
- gritzko/swarm - 010-setrm.ron
- gritzko/swarm - 011-react.ron
- gritzko/swarm - 012-local-uuids.ron
- gritzko/swarm - 013-once.ron
- gritzko/swarm - 014-gql-subs.ron
- gritzko/swarm - 016-gql-query.ron
- gritzko/swarm - 017-ensure-directive.ron
- gritzko/swarm - 018-react-graphql.ron
- gritzko/swarm - 019-pending.ron
- gritzko/swarm - 020-empty-ack.ron
- gritzko/swarm - 021-gql-empty-ack.ron
- gritzko/swarm - gql-mutation.ron
- nuxdie/sflovers - Goulart.Ron
- nuxdie/sflovers - Hubbard.L.Ron
</details>
<br>

Being fully honest, my driver for this PR is removing `.ron` from the list of markdown extensions.
Rendering a text-serialization format as if it were markdown makes the GitHub preview unusable.
It's a lossy translation.

As hard as I've tried, [setting `.gitattributes` `linguist-language` _does not change that_](https://github.com/CAD97/ron-display/blob/master/.gitattributes).
The preview GitHub served is still markdown rendered.

File: samples/ronn/bunyan.1.ronn [hunk 2]
```diff
@@ -0,0 +1,195 @@
+# bunyan(1) -- filter and pretty-print Bunyan log file content
```
Comment (pchaigno) [inline samples/ronn/bunyan.1.ronn]: The sample files should be in `samples/Markdown/`. (It's the cause of the of the Travis CI error.)

Comment (CAD97) [reply samples/ronn/bunyan.1.ronn]: fixed 👍

**Comment by kvark** [comment]
@pchaigno ping! we are sort of suffering here :)
e.g. unable to see or get a link for a line in https://github.com/gfx-rs/gfx/blob/master/reftests/scenes/basic.ron

File: lib/linguist/languages.yml [hunk 3]
```diff
@@ -2659,7 +2659,7 @@ Markdown:
   - ".mkd"
   - ".mkdn"
   - ".mkdown"
-  - ".ron"
```
[the remark this answers is not in the document]
Comment (pchaigno) [reply lib/linguist/languages.yml]: @lildude Did you get a chance to look into this?

**Comment by CAD97** [comment]
GitHub Linguist changes require a sign off from a GitHub staffer so I think we're waiting on @lildude to pop by and make a decision.

This patch was made a few days after a deploy IIRC, so we'll have a bit to wait for the next deploy whether this merges now or just before the next deploy.

File: lib/linguist/languages.yml (hunk 3 above)
[the remark this answers is not in the document]
Comment (lildude) [reply lib/linguist/languages.yml]: Been AFK for a while so only looking now.

I've looked into this more closely and I think we can safely drop `.ron` from Markdown. So few results are returned, we'd never include them in a new addition today and as pointed out by @CAD97, these are mostly from https://github.com/mustache/mustache which is still using the old `.ron` extension.

That said, I think we should be good citizens and send mustache/mustache a PR to rename these files.

[the remark this answers is not in the document]
Comment (pchaigno) [reply lib/linguist/languages.yml]: > That said, I think we should be good citizens and send mustache/mustache a PR to rename these files.

:+1: 

@CAD97 Do you want to open that pull request? I can do it, but you're certainly more knowledgeable than I on the matter.

**Comment by CAD97** [comment]
Now that the change is live, [I have sent a PR to mustache/mustache to rename their ronn files from `.ron` to `.ronn`](https://github.com/mustache/mustache/pull/242). Thanks to everyone for working with us! 🎉

**Pull request merged**